### PR TITLE
Add documentation of the vulnerability detector tests

### LINKED ISSUE #799: Vulnerability detector: Tests documentation
Hello team,

The CICD team is working on new integration tests for the vulnerability detector module and it would be necessary to add documentation of all checks performed in these tests so that the team knows what cases are being tested and how it's being done.

This documentation will be included in this repository via readme files.

Best regards.

### PR DESCRIPTION
**Pull request by jmv74211**
Hello team,

This PR adds the documentation of the vulnerability detector tests.

This documentation has been structured using `README` files in the vulnerability detector directory of the `QA-repository`, and it has been designed in a way that you can browse the information through the different links.

For more information, see https://github.com/wazuh/wazuh-qa/tree/799-tests-documentation/tests/integration/test_vulnerability_detector

Closes #799 

Best regards.

### CODE REVIEW

File: tests/integration/test_vulnerability_detector/readme.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+#### Test feeds - GENERIC
+
+- **[test_invalid_type_custom_feeds](./doc/test_feeds/generic/test_invalid_type_custom_feeds.md)**: Tests that are
+based on importing files of several different types (`.mp3`, `.jpg`, `.pdf` ...) as custom feed, and observing the
+response of vulnerability detector.
+
+- **[test_validate_feed_content](./doc/test_feeds/generic/test_validate_feed_content.md)**: Tests that are based on
+downloading the feeds of all providers and verifying that the format of each feed is as expected and that the
+content of the feed is also `XML` or `JSON` parseable.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/readme.md:103]:
```suggestion
- **[test_validate_feed_content](./doc/test_feeds/generic/test_validate_feed_content.md)**: Tests that download the feeds of all providers, verify the format of each feed is as expected and their content is also `XML` or `JSON` parseable.
```

File: tests/integration/test_vulnerability_detector/readme.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+#### Test feeds - RHEL
+
+- **[test_extra_fields_redhat_feed](./doc/test_feeds/redhat/test_extra_fields_redhat_feed.md)**: Set of tests that are
+based on checking the behavior of vulnerability detector when there is any extra field in the feed.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/readme.md:108]:
```suggestion
- **[test_extra_fields_redhat_feed](./doc/test_feeds/redhat/test_extra_fields_redhat_feed.md)**: Set of tests that check the behavior of vulnerability detector when there is any extra field in the feed.
```

File: tests/integration/test_vulnerability_detector/readme.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+- **[test_invalid_syntax_redhat_feed](./doc/test_feeds/redhat/test_invalid_syntax_redhat_feed.md)**: Set of tests that
+are based on checking the behavior of vulnerability detector when the feed has some kind of syntactic error: it doesn't
+have a closing field, it's missing some character ...
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/readme.md:112]:
```suggestion
- **[test_invalid_syntax_redhat_feed](./doc/test_feeds/redhat/test_invalid_syntax_redhat_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when the feed has some kind of syntactic error.
```

File: tests/integration/test_vulnerability_detector/readme.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+- **[test_invalid_values_redhat_feed](./doc/test_feeds/redhat/test_invalid_values_redhat_feed.md)**: Set of tests that
+are based on checking the behavior of vulnerability detector when the value of a field is not correct: invalid type,
+strange characters ...
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/readme.md:116]:
```suggestion
- **[test_invalid_values_redhat_feed](./doc/test_feeds/redhat/test_invalid_values_redhat_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when the value of a field is incorrect.
```

File: tests/integration/test_vulnerability_detector/readme.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+- **[test_missing_fields_redhat_feed](./doc/test_feeds/redhat/test_missing_fields_redhat_feed.md)**: Set of tests that
+are based on checking the behavior of vulnerability detector when any field is missing from the feed.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/readme.md:119]:
```suggestion
- **[test_missing_fields_redhat_feed](./doc/test_feeds/redhat/test_missing_fields_redhat_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when any field is missing from the feed.
```

File: tests/integration/test_vulnerability_detector/readme.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+#### Test feeds - CANONICAL
+
+- **[test_extra_tags_canonical_feed](./doc/test_feeds/canonical/test_extra_tags_canonical_feed.md)**:  Set of tests
+that are based on checking the behavior of vulnerability detector when there is any extra tag in the feed.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/readme.md:124]:
```suggestion
- **[test_extra_tags_canonical_feed](./doc/test_feeds/canonical/test_extra_tags_canonical_feed.md)**: Set of tests
that check the behavior of Vulnerability Detector when there is any extra tag in the feed.
```

File: tests/integration/test_vulnerability_detector/readme.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+- **[test_invalid_syntax_canonical_feed](./doc/test_feeds/canonical/test_invalid_syntax_canonical_feed.md)**: Set of
+tests that are based on checking the behavior of vulnerability detector when the feed has some kind of syntactic error:
+it doesn't have a closing tag, it's missing some character ...
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/readme.md:128]:
```suggestion
- **[test_invalid_syntax_canonical_feed](./doc/test_feeds/canonical/test_invalid_syntax_canonical_feed.md)**: Set of
tests that check the behavior of Vulnerability Detector when the feed has some kind of syntactic error.
```

File: tests/integration/test_vulnerability_detector/readme.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+- **[test_invalid_values_canonical_feed](./doc/test_feeds/canonical/test_invalid_values_canonical_feed.md)**: Set of
+tests that are based on checking the behavior of vulnerability detector when the value of a tag is not correct:
+invalid type, strange characters ...
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/readme.md:132]:
```suggestion
- **[test_invalid_values_canonical_feed](./doc/test_feeds/canonical/test_invalid_values_canonical_feed.md)**: Set of
tests that check the behavior of Vulnerability Detector when the value of a tag is incorrect.
```

File: tests/integration/test_vulnerability_detector/readme.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+
+- **[test_missing_tags_canonical_feed](./doc/test_feeds/canonical/test_missing_tags_canonical_feed.md)**: Set of tests
+that are based on checking the behavior of vulnerability detector when any tag is missing from the feed.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/readme.md:135]:
```suggestion
- **[test_missing_tags_canonical_feed](./doc/test_feeds/canonical/test_missing_tags_canonical_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when any tag is missing from the feed.
```

File: tests/integration/test_vulnerability_detector/readme.md [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+#### Test feeds - DEBIAN
+
+- **[test_extra_tags_debian_feed](./doc/test_feeds/debian/test_extra_tags_debian_feed.md)**: Set of tests that are
+based on checking the behavior of vulnerability detector when there is any extra tag in the feed.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/readme.md:140]:
```suggestion
- **[test_extra_tags_debian_feed](./doc/test_feeds/debian/test_extra_tags_debian_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when there is any extra tag in the feed.
```

File: tests/integration/test_vulnerability_detector/readme.md [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+
+- **[test_invalid_syntax_debian_feed](./doc/test_feeds/debian/test_invalid_syntax_debian_feed.md)**: Set of tests that
+are based on checking the behavior of vulnerability detector when the feed has some kind of syntactic error: it doesn't
+have a closing tag, it's missing some character ...
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/readme.md:144]:
```suggestion
- **[test_invalid_syntax_debian_feed](./doc/test_feeds/debian/test_invalid_syntax_debian_feed.md)**: Set of tests check the behavior of Vulnerability Detector when the feed has some kind of syntactic error.
```

File: tests/integration/test_vulnerability_detector/readme.md [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+
+- **[test_invalid_values_debian_feed](./doc/test_feeds/debian/test_invalid_values_debian_feed.md)**: Set of tests that
+are based on checking the behavior of vulnerability detector when the value of a tag is not correct: invalid type,
+strange characters ...
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/readme.md:148]:
```suggestion
- **[test_invalid_values_debian_feed](./doc/test_feeds/debian/test_invalid_values_debian_feed.md)**: Set of tests that check the behavior of vulnerability detector when the value of a tag is not correct.
```

File: tests/integration/test_vulnerability_detector/readme.md [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+
+- **[test_missing_tags_debian_feed](./doc/test_feeds/debian/test_missing_tags_debian_feed.md)**: Set of tests that are
+based on checking the behavior of vulnerability detector when any tag is missing from the feed.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/readme.md:151]:
```suggestion
- **[test_missing_tags_debian_feed](./doc/test_feeds/debian/test_missing_tags_debian_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when any tag is missing from the feed.
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 14]
```diff
@@ -0,0 +1,88 @@
+# Test feeds
+
+## Objective
+
+The purpose of testing imported feeds is to check the behavior of the vulnerability detector when importing feeds that
+may have some kind of error.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:6]:
```suggestion
The purpose of testing imported feeds is to check the behavior of Vulnerability Detector when importing feeds that may have some kind of error.
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+## General info
+
+|Tier | Number of tests | Time spent |
+|:--:|:--:|:--:|
+| 1-2 | 1134 | 2h 53m |
+
+## Expected behavior
+
+The expected behavior is to **discard** those vulnerabilities whose content is malformed making vulnerability detector
+**not fail** for any related reason.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:17]:
```suggestion
The expected behavior is to **discard** those vulnerabilities whose content is malformed making Vulnerability Detector **not fail** for it.
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+
+## Testing
+
+The tests can be classified into:
+
+- **Generics**: Generic tests that are carried out on the feeds, regardless of the provider.
+- **Providers**: Specific tests of each provider in which the feed is altered and the result is evaluated.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:24]:
```suggestion
- **Generics**: tests that are carried out on the feeds, regardless of the provider.
- **Providers**: specific tests of each provider where the feed is modified and the result is evaluated.
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+
+### Generic tests
+
+- **[test_download_feeds](../doc/test_feeds/generic/test_download_feeds.md)**: Tests that are based on downloading the
+different feeds (redhat, canonical, debian and NVD) and looking for the message confirming that the feeds have been
+downloaded and imported correctly.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:30]:
```suggestion
- **[test_download_feeds](../doc/test_feeds/generic/test_download_feeds.md)**: Tests that download the different feeds (Redhat, Canonical, Debian, and NVD) and check the confirmation message saying the feeds have been correctly downloaded and imported.
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+
+- **[test_invalid_type_custom_feeds](../doc/test_feeds/generic/test_invalid_type_custom_feeds.md)**: Tests that are
+based on importing files of several different types (`.mp3`, `.jpg`, `.pdf` ...) as custom feed, and observing the
+response of vulnerability detector.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:34]:
```suggestion
- **[test_invalid_type_custom_feeds](../doc/test_feeds/generic/test_invalid_type_custom_feeds.md)**: Tests that importing files of several different types (`.mp3`, `.jpg`, `.pdf` ...) as custom feed, and check the response of Vulnerability Detector.
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 19]
```diff
@@ -0,0 +1,88 @@
...
+
+- **[test_validate_feed_content](../doc/test_feeds/generic/test_validate_feed_content.md)**: Tests that are based on
+downloading the feeds of all providers and verifying that the format of each feed is as expected and that the
+content of the feed is also `XML` or `JSON` parseable.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:38]:
```suggestion
- **[test_validate_feed_content](../doc/test_feeds/generic/test_validate_feed_content.md)**: Tests that download the feeds of all the providers and verify their format is as expected and their content is also `XML` or `JSON` parseable.
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+
+### Providers tests
+
+#### Red Hat
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:42]:
```suggestion
#### RedHat
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 21]
```diff
@@ -0,0 +1,88 @@
...
+
+- **[test_extra_fields_redhat_feed](../doc/test_feeds/redhat/test_extra_fields_redhat_feed.md)**: Set of tests that are
+based on checking the behavior of vulnerability detector when there is any extra field in the feed.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:45]:
```suggestion
- **[test_extra_fields_redhat_feed](../doc/test_feeds/redhat/test_extra_fields_redhat_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when there is any extra field in the feed.
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 22]
```diff
@@ -0,0 +1,88 @@
...
+
+- **[test_invalid_syntax_redhat_feed](../doc/test_feeds/redhat/test_invalid_syntax_redhat_feed.md)**: Set of tests that
+are based on checking the behavior of vulnerability detector when the feed has some kind of syntactic error: it doesn't
+have a closing field, it's missing some character ...
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:49]:
```suggestion
- **[test_invalid_syntax_redhat_feed](../doc/test_feeds/redhat/test_invalid_syntax_redhat_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when the feed has syntactic errors.
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 23]
```diff
@@ -0,0 +1,88 @@
...
+
+- **[test_invalid_values_redhat_feed](../doc/test_feeds/redhat/test_invalid_values_redhat_feed.md)**: Set of tests that
+are based on checking the behavior of vulnerability detector when the value of a field is not correct: invalid type,
+strange characters ...
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:53]:
```suggestion
- **[test_invalid_values_redhat_feed](../doc/test_feeds/redhat/test_invalid_values_redhat_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when the value of a field is not correct.
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 24]
```diff
@@ -0,0 +1,88 @@
...
+
+- **[test_missing_fields_redhat_feed](../doc/test_feeds/redhat/test_missing_fields_redhat_feed.md)**: Set of tests that
+are based on checking the behavior of vulnerability detector when any field is missing from the feed.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:56]:
```suggestion
- **[test_missing_fields_redhat_feed](../doc/test_feeds/redhat/test_missing_fields_redhat_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when any field is missing from the feed.
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 25]
```diff
@@ -0,0 +1,88 @@
...
+
+#### Canonical
+
+- **[test_extra_tags_canonical_feed](../doc/test_feeds/canonical/test_extra_tags_canonical_feed.md)**:  Set of tests
+that are based on checking the behavior of vulnerability detector when there is any extra tag in the feed.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:61]:
```suggestion
- **[test_extra_tags_canonical_feed](../doc/test_feeds/canonical/test_extra_tags_canonical_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when there is any extra tag in the feed.
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 26]
```diff
@@ -0,0 +1,88 @@
...
+
+- **[test_invalid_syntax_canonical_feed](../doc/test_feeds/canonical/test_invalid_syntax_canonical_feed.md)**: Set of
+tests that are based on checking the behavior of vulnerability detector when the feed has some kind of syntactic error:
+it doesn't have a closing tag, it's missing some character ...
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:65]:
```suggestion
- **[test_invalid_syntax_canonical_feed](../doc/test_feeds/canonical/test_invalid_syntax_canonical_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when the feed has some kind of syntactic error.
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 27]
```diff
@@ -0,0 +1,88 @@
...
+
+- **[test_invalid_values_canonical_feed](../doc/test_feeds/canonical/test_invalid_values_canonical_feed.md)**: Set of
+tests that are based on checking the behavior of vulnerability detector when the value of a tag is not correct:
+invalid type, strange characters ...
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:69]:
```suggestion
- **[test_invalid_values_canonical_feed](../doc/test_feeds/canonical/test_invalid_values_canonical_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when the value of a tag is not correct.
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 28]
```diff
@@ -0,0 +1,88 @@
...
+
+- **[test_missing_tags_canonical_feed](../doc/test_feeds/canonical/test_missing_tags_canonical_feed.md)**: Set of tests
+that are based on checking the behavior of vulnerability detector when any tag is missing from the feed.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:72]:
```suggestion
- **[test_missing_tags_canonical_feed](../doc/test_feeds/canonical/test_missing_tags_canonical_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when any tag is missing from the feed.
```

File: tests/integration/test_vulnerability_detector/test_feeds/readme.md [hunk 29]
```diff
@@ -0,0 +1,88 @@
...
+- **[test_invalid_syntax_debian_feed](../doc/test_feeds/debian/test_invalid_syntax_debian_feed.md)**: Set of tests that
+are based on checking the behavior of vulnerability detector when the feed has some kind of syntactic error: it doesn't
+have a closing tag, it's missing some character ...
+
+- **[test_invalid_values_debian_feed](../doc/test_feeds/debian/test_invalid_values_debian_feed.md)**: Set of tests that
+are based on checking the behavior of vulnerability detector when the value of a tag is not correct: invalid type,
+strange characters ...
+
+- **[test_missing_tags_debian_feed](../doc/test_feeds/debian/test_missing_tags_debian_feed.md)**: Set of tests that are
+based on checking the behavior of vulnerability detector when any tag is missing from the feed.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/readme.md:88]:
```suggestion
- **[test_extra_tags_debian_feed](../doc/test_feeds/debian/test_extra_tags_debian_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when there is any extra tag in the feed.

- **[test_invalid_syntax_debian_feed](../doc/test_feeds/debian/test_invalid_syntax_debian_feed.md)**: Set of tests that checking the behavior of Vulnerability Detector when the feed has some kind of syntactic error.

- **[test_invalid_values_debian_feed](../doc/test_feeds/debian/test_invalid_values_debian_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when the value of a tag is not correct.

- **[test_missing_tags_debian_feed](../doc/test_feeds/debian/test_missing_tags_debian_feed.md)**: Set of tests that check the behavior of Vulnerability Detector when any tag is missing from the feed.
```

File: tests/integration/test_vulnerability_detector/test_scan_results/readme.md [hunk 30]
```diff
@@ -0,0 +1,45 @@
...
+that are based on mocking ubuntu systems and inserting custom vulnerabilities and vulnerable packages to see if
+vulnerability detector generates the vulnerability alerts from Canonical provider feed.
+
+- **[test_scan_nvd_feed](../doc/test_scan_results/test_scan_nvd_feed.md)**: Tests that are based on mocking Red Hat,
+Canonical, Debian and Windows systems, and inserting custom vulnerabilities and vulnerable packages to see if
+vulnerability detector generates the vulnerability alerts from NVD feed.
+
+- **[test_scan_providers_and_nvd_feed](../doc/test_scan_results/test_scan_providers_and_nvd_feed.md)**: Tests that are
+based on mocking Red Hat, Canonical and Debian systems, and inserting custom vulnerabilities and vulnerable packages to
+see if vulnerability detector generates the vulnerability alerts from NVD and providers feed.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_scan_results/readme.md:45]:
```suggestion
- **[test_debian_inventory_debian_feed](../doc/test_scan_results/test_debian_inventory_debian_feed.md)**: Tests that mock Debian systems and insert custom vulnerabilities and vulnerable packages to check if Vulnerability Detector generates alerts from Debian provider feed.

- **[test_redhat_inventory_redhat_feed](../doc/test_scan_results/test_redhat_inventory_redhat_feed.md)**: Tests that mock RedHat systems and insert custom vulnerabilities and vulnerable packages to check if Vulnerability Detector generates alerts from RedHat provider feed.

- **[test_ubuntu_inventory_canonical_feed](../doc/test_scan_results/test_ubuntu_inventory_canonical_feed.md)**: Tests
that mock Ubuntu systems and insert custom vulnerabilities and vulnerable packages to see if Vulnerability Detector generates alerts from Canonical provider feed.

- **[test_scan_nvd_feed](../doc/test_scan_results/test_scan_nvd_feed.md)**: Tests that mock RedHat, Ubuntu, Debian and Windows systems, and insert custom vulnerabilities and vulnerable packages to check if Vulnerability Detector generates alerts from NVD feed.

- **[test_scan_providers_and_nvd_feed](../doc/test_scan_results/test_scan_providers_and_nvd_feed.md)**: Tests that mock RedHat, Ubuntu and Debian systems, and insert custom vulnerabilities and vulnerable packages to check if Vulnerability Detector generates alerts from NVD and providers feed.
```

File: tests/integration/test_vulnerability_detector/doc/test_feeds/redhat/test_invalid_values_redhat_feed.md [hunk 31]
```diff
@@ -0,0 +1,77 @@
...
+```
+
+is assigned the following values:
+
+```
+[None, "", "dummy value", 12345, ['1', '2', '3', '4', '5'], "ñ", "テスト",
+ "ИСПЫТАНИЕ", "测试", "اختبار"]
+```
+
+For instance:
```
[the remark this answers is not in the document]
Comment (jmv74211) [reply tests/integration/test_vulnerability_detector/doc/test_feeds/redhat/test_invalid_values_redhat_feed.md]: Updated in 1b482b8f45644cac1c6aea67e6aec428b66e3b41

File: tests/integration/test_vulnerability_detector/doc/test_feeds/debian/test_missing_tags_debian_feed.md [hunk 32]
```diff
@@ -0,0 +1,69 @@
...
+
+The feed will not be imported if:
+
+- There is a key tag missing:
+
+  ```
+  key_tags = ['definitions', 'definition', 'metadata', 'reference', 'criteria']
+  ```
+
+**Xfailing** (https://github.com/wazuh/wazuh/issues/5322) if there is one of the following tags missing:
```
[the remark this answers is not in the document]
Comment (jmv74211) [reply tests/integration/test_vulnerability_detector/doc/test_feeds/debian/test_missing_tags_debian_feed.md]: Updated in 1b482b8f45644cac1c6aea67e6aec428b66e3b41

File: tests/integration/test_vulnerability_detector/doc/test_feeds/debian/test_invalid_values_debian_feed.md [hunk 33]
```diff
@@ -0,0 +1,72 @@
...
+
+```
+[None, "", "dummy value", 12345, ['1', '2', '3', '4', '5'], "ñ", "テスト",
+ "ИСПЫТАНИЕ", "测试", "اختبار"]
+```
+
+For instance:
+
+```
+<definitions>None</definitions>
```
[the remark this answers is not in the document]
Comment (jmv74211) [reply tests/integration/test_vulnerability_detector/doc/test_feeds/debian/test_invalid_values_debian_feed.md]: Updated in 1b482b8f45644cac1c6aea67e6aec428b66e3b41

**Review by BraulioV — DISMISSED** [review]

**Comment by BraulioV** [comment]
This is closed in favor of [#1066](https://github.com/wazuh/wazuh-qa/pull/1066)

**Pull request closed without merging**